src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_NavigateTo.cs

```diff
@@ -21,15 +21,15 @@ internal partial class CodeAnalysisService : IRemoteNavigateToSearchService
 
                     var project = solution.GetDocument(documentId);
                     var result = await AbstractNavigateToSearchService.SearchDocumentInCurrentProcessAsync(
-                        project, searchPattern, token).ConfigureAwait(false);
+                        project, searchPattern, kinds.ToImmutableHashSet(), token).ConfigureAwait(false);
```

Comment:
note: HashSet is faster.  Consider potentially just using that given the hotspotness of checking all members.

Reply:
If perf is a problem I'll probably switch to an enum set (array indexed by `DeclaredSymbolInfoKind`).

Reply:
Sure.  That would work fine for me as well.  But i think it would be good to have that from the start.  Doing potentially 1m+ interface dispatches, hashes, and equality checks for all these strings just seems like something we'd want to avoid from the start.

Note: i had little test apps i wrote a while back to measure perf here when i would make pattern-matching changes.  Unfortunately, i checked, and i don't seem to have it on any of my machines.  But creating one should not be too difficult.  I've found that roslyn in a reasonable test suite for these sorts of measurements.

Reply:
The immutable set is an intermediate structure that appears to be infrequently allocated relative to the number of times the final enum set is accessed. Seems like a reasonable intermediate data structure here that would throw if someone attempted to write to it.

Reply:
I'm fine with that with the new form (that optimizes accesses to be reasonably fast) :)